mbed Enabled certificate for the VBLUno51 board

iotvietmember opened the pull request:
## Description
This PR add new BOARD_ID for the VBLUno51 board. 

The mbed Enabled process for the VBLUno51 board 


## Status
**READY**

## Migrations
NO

Signed-off-by: iotvietmember <robotden@gmail.com>

adbridge commented:
@iotvietmember please remove the merge commit and do a rebase instead. Thanks

iotvietmember commented:
@adbridge I have just fixed those problem. Thanks

adbridge commented:
@iotvietmember Thanks :)

adbridge commented:
/morph test

iotvietmember commented:
@adbridge I saw the good results

theotherjimmy commented:
@iotvietmember This PR is on hold until we get 10 samples. See https://www.mbed.com/en/about-mbed/mbed-enabled/#Program_Requirements for more details.

iotvietmember commented:
@theotherjimmy  We have sent 10 samples of the VBLUno51 board for mbed team in Japan on 08 June 2017.
[Tracking](https://www.tnt.com/express/en_vn/site/shipping-tools/tracking.html?cons=406266022)

And the following file is apply form for mbed Enabled [Apply form](https://github.com/ARMmbed/mbed-os/files/1137369/2_ABaOnuehyCuYzuW1DveD5zwLJobOo_jwYXkCCmBSn2iC-oOC2P8x3FFQ19FvoA.pdf)

0xc0170 commented:
> @theotherjimmy We have sent 10 samples of the VBLUno51 board for mbed team in Japan on 08 June 2017.

cc @toyowata

iotvietmember commented:
@toyowata Would you please check the mbed Enabled progress for the VBLUno51 board. Many thanks

toyowata commented:
I sent the board to Austin/TX team from Japan two days ago.  Test engineer will be assigned in that team for the VBLUno51 board.

iotvietmember commented:
@toyowata @0xc0170 @theotherjimmy  Thanks for your support and hope that we will be received the good news from your soon.

0xc0170 commented:
There's a conflict now.

More important, 

> I sent the board to Austin/TX team from Japan two days ago. Test engineer will be assigned in that team for the VBLUno51 board.
 
@toyowata Any update on this one. @studavekar  ? This is however connected to the enabled program, this PR should be good to. 

@iotvietmember once the conflict is resolved, can you share with us the testing ? That you did for mbed OS 5 (as this enables this version).

iotvietmember commented:
@0xc0170 Yes, of course. I am always ready share with you the testing for VBLUno51 board on ARMmbed-os. Could the VBLUno51 board receive mbed Enabled 5.0 or 2.0 version? We hope that our board could receive mbed Enabled 5.0 version with free charge. If it can not, we hope that our board will receive mbed Enabled 2.0 soon.
Do you think that I should send to you the testing for mbed Enabled 5.0 or 2.0?
Many thanks?

0xc0170 commented:
@iotvietmember Please rebase to resolve the conflicts, and share the testing with us here

iotvietmember commented:
Hi @0xc0170 I have rebased to resolve conflicts. Beside that, I also add detect_code field for the VBLUno51 board config - C006.

iotvietmember commented:
@0xc0170  I also did the testing for VBLUno51 board on mbed-os. For details, please see below.

1. Enviroment
- OS: Microsoft Windows 7 Professional 64 bit
- mbed-os version: `5.5.6` (clone from master branch of mbed-os on 05-Sep-2017)
- Update `new DAPLink firmware` with `BOARD_ID = 0xC006` 
https://drive.google.com/open?id=0B0xLV-EhFkn4N1JydXNBcXpibVk

2. Create `test` directory, contains:
![1](https://user-images.githubusercontent.com/15735613/30221749-b57574d0-94ee-11e7-9383-94303869f134.PNG)

3. `config` files:
- `test_spec.json`
```
{
    "targets": {
        "VBLUNO51": ["GCC_ARM"]
    }
}
```
- `muts_all.json`
```
{
    "1" : {"mcu": "VBLUNO51",
           "port":"COM72",
           "disk":"F:\\",
           "peripherals": []
    }
}
```

4. `mbed-os-5.5.6` directory
![2](https://user-images.githubusercontent.com/15735613/30222087-25ef9302-94f0-11e7-97d1-6726946430fb.PNG)

5. `run_mbed_testsuite.bat` script file
```
python mbed-os-5.5.6\tools\singletest.py --version
python mbed-os-5.5.6\tools\singletest.py -i config\test_spec.json -M config\muts_all.json -t --report-build output\REPORT_BUILD.txt --report-text output\REPORT_TEXT.txt
pause
```
6. **TEST RESULTS**
```
+--------+----------+-----------+-------------+---------------------------------------+--------------+---------+
| Result | Target   | Toolchain | Test ID     | Test Description                      | Elapsed Time | Timeout |
+--------+----------+-----------+-------------+---------------------------------------+--------------+---------+
| OK     | VBLUNO51 | GCC_ARM   | DTCT_1      | Simple detect test                    |     0.76     |    10   |
| OK     | VBLUNO51 | GCC_ARM   | EXAMPLE_1   | /dev/null                             |     3.75     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_10     | Hello World                           |     0.64     |    5    |
| OK     | VBLUNO51 | GCC_ARM   | MBED_11     | Ticker Int                            |    11.64     |    15   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_12     | C++                                   |     1.63     |    10   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_16     | RTC                                   |     4.87     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_2      | stdio                                 |     1.03     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_23     | Ticker Int us                         |    11.64     |    15   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_24     | Timeout Int us                        |    12.74     |    15   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_25     | Time us                               |    11.87     |    15   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_26     | Integer constant division             |     1.65     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_34     | Ticker Two callbacks                  |    11.64     |    15   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_37     | Serial NC RX                          |     7.25     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_38     | Serial NC TX                          |     6.28     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_A1     | Basic                                 |     1.63     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_A21    | Call function before main (mbed_main) |     1.71     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_A9     | Serial Echo at 115200                 |     1.87     |    20   |
| OK     | VBLUNO51 | GCC_ARM   | MBED_BUSOUT | BusOut                                |     2.57     |    30   |
+--------+----------+-----------+-------------+---------------------------------------+--------------+---------+
Result: 18 OK
```

[REPORT_TEXT.txt](https://github.com/ARMmbed/mbed-os/files/1288578/REPORT_TEXT.txt)

[REPORT_BUILD.txt](https://github.com/ARMmbed/mbed-os/files/1288579/REPORT_BUILD.txt)

Many thanks for your supports!

iotvietmember commented:
@0xc0170 Could you please check the above comment? Many thanks

iotvietmember commented:
@0xc0170 @theotherjimmy Could you please check the above commit soon? Many thanks

iotvietmember commented:
@theotherjimmy I understand that the PR title does not match,but @0xc0170 want me to rebase and resolve conflicts. And the VBLUno51 PR on DAPLink, pyOCD, mbed-ls is linked to this PR.

theotherjimmy commented:
@iotvietmember They are linked by number, not by name. You are welcome to change the title.

iotvietmember commented:
@theotherjimmy  I have just changed the PR title to "mbed Enabled certificate for the VBLUno51 board"

iotvietmember commented:
@theotherjimmy I do not understand which you want to share

iotvietmember commented:
@0xc0170 @theotherjimmy Could you please help me solve the problem?

theotherjimmy commented:
@iotvietmember The Travis CI failure is related to the base of this branch. We resolved that issue a little while back, so a rebase onto the current master would bring you up to date and allow Travis CI to pass.

iotvietmember commented:
@theotherjimmy I have just rebase and fix the problem

theotherjimmy commented:
Awesome! Travis is happy now.

iotvietmember commented:
@theotherjimmy I hope that the mbed Enabled process for the VBLUno51 board will finish soon. Many thanks for your support!

0xc0170 commented:
@toyowata Still do not merge state for this one, or ungated and all verified?

toyowata commented:
@0xc0170 Sorry, I am not handling this new platform (I have sent all board to US).

iotvietmember commented:
@0xc0170 Many thanks

iotvietmember commented:
@0xc0170 Do you have a problem with the VBLUno51 board? I hope that I can help you.

0xc0170 commented:
/morph export-build

0xc0170 commented:
/morph test

theotherjimmy commented:
Waiting on review from DevX

senthilr commented:
The platform page is being populated, a final sanity check and stamp of approval from Marketing team to follow. 
Then this would be good for merge.

iotvietmember commented:
@theotherjimmy , @senthilr Many thanks for your support

theotherjimmy commented:
@0xc0170 I did not see a mention back from @senthilr stating that this is ready for merge.

0xc0170 commented:
> @0xc0170 I did not see a mention back from @senthilr stating that this is ready for merge.
 
@senthilr Please proceed

senthilr commented:
still WIP. sorry this is taking a while to process.

0xc0170 commented:
> still WIP. sorry this is taking a while to process.

Bump for update :-)

iotvietmember commented:
Hi @senthilr, 

The platform page for the VBLUNO51 board is ready at `https://os.mbed.com/platforms/VBLUNO51/`.
Could you please resend the permission letter to me soon. 
Many thanks for your support.

iotvietmember commented:
Hi @0xc0170, @theotherjimmy, @senthilr : I have just rebased the commit. I added some lines in targets.json file. The VBLUno51 board needs to be disabled UART Hardware flow control feature, compatible with `nrf51822` target on mbed Online Compiler.

The DAPLINK firmware needs to upgraded up `fw_daplink_vbluno51_release_1.0.1.bin`
Instructions at 
[https://os.mbed.com/teams/VNGIoTLab/wiki/vbluno51-interface-firmware](https://os.mbed.com/teams/VNGIoTLab/wiki/vbluno51-interface-firmware)

0xc0170 commented:
/morph build

adbridge commented:
@0xc0170, @theotherjimmy, @senthilr  Could you please confirm you are happy with the changes ?

studavekar commented:
/morph build

studavekar commented:
/morph export-build

iotvietmember commented:
What is the problem?

0xc0170 commented:
> What is the problem?

We are currently investigating the K64F failures, not related to this pull request.

studavekar commented:
Looks like master is all green starting fresh build 

## Triggering build 

/morph build

iotvietmember commented:
What do I need to do?

theotherjimmy commented:
@iotvietmember I think this is waiting on @senthilr's review.

c1728p9 commented:
Hi @iotvietmember, I took a look at why this was failing ```tests-mbed_drivers-ticker``` on gcc. Its because UART data is being lost during the test. If flow control is enabled in DAPLink and on the VBLUNO51 then this test passes reliably.

>  The VBLUno51 board needs to be disabled UART Hardware flow control feature, compatible with nrf51822 target on mbed Online Compiler.

You mentioned that flow control needed to be turned off due to the nrf51822. The VBLUno51 has an nrf51822 on it and it works with flow control. By nrf51822 do you mean a board other than the VBLUno51? If so what board?

adbridge commented:
@iotvietmember Could you please address address @c1728p9 question ?

cmonr commented:
Are we still waiting on DevX/@senthilr's ok, since the MBed page for the hardware is live?

@iotvietmember Ping on @c1728p9's question.

iotvietmember commented:
@cmonr @adbridge @c1728p9 My board is the VBLUno51 and we hope that we can use 'VBLUNO51' as target name on mbed. 
On hardware, the VBLUno51 has UART hardware flow control.
1. When I created the https://os.mbed.com/platforms/VBLUNO51/, A notice in the upper right corner with content 
"To compile a program for this board using Mbed CLI, use **nrf51822** as the target name."
So I have to cancel the HWFC UART feature on the VBLUno51.

2. After that, you modified
"To compile a program for this board using Mbed CLI, use **VBLUNO51** as the target name."
This means that we can use VBLUNO51 as the name of the board on the mbed. This means that the UART HWFC feature is used (fw_daplink_vbluno51_release_1.0.0.bin
https://os.mbed.com/teams/VNGIoTLab/wiki/vbluno51-interface-firmware)

Many thanks

cmonr commented:
@iotvietmember Thanks for that clarification!
